pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/loader/bloomfilter/BloomFilterHandler.java

```diff
@@ -54,52 +55,59 @@
 public class BloomFilterHandler implements IndexHandler {
   private static final Logger LOGGER = LoggerFactory.getLogger(BloomFilterHandler.class);
 
-  private final File _indexDir;
-  private final SegmentMetadataImpl _segmentMetadata;
-  private final SegmentDirectory.Writer _segmentWriter;
+  private final SegmentMetadata _segmentMetadata;
   private final Map<String, BloomFilterConfig> _bloomFilterConfigs;
-  private final BloomFilterCreatorProvider _indexCreatorProvider;
 
-  public BloomFilterHandler(File indexDir, SegmentMetadataImpl segmentMetadata, IndexLoadingConfig indexLoadingConfig,
-      SegmentDirectory.Writer segmentWriter, BloomFilterCreatorProvider indexCreatorProvider) {
-    _indexDir = indexDir;
-    _segmentWriter = segmentWriter;
+  public BloomFilterHandler(SegmentMetadata segmentMetadata, IndexLoadingConfig indexLoadingConfig) {
     _segmentMetadata = segmentMetadata;
     _bloomFilterConfigs = indexLoadingConfig.getBloomFilterConfigs();
-    _indexCreatorProvider = indexCreatorProvider;
   }
 
   @Override
-  public void updateIndices()
+  public boolean needUpdateIndices(SegmentDirectory.Reader segmentReader) {
+    Set<String> columnsToAddBF = new HashSet<>(_bloomFilterConfigs.keySet());
```

Comment:
Mostly lgtm.
Minor comment: can this code be moved into a util? any updates needed to be made to this logic in the future will stay in one place

Reply:
actually, think I just need an equality check here